src/main/java/edu/cornell/kfs/fp/batch/service/impl/AccountingXmlDocumentDownloadAttachmentServiceImpl.java

```diff
@@ -36,6 +36,11 @@
     @Override
     public Attachment createAttachmentFromBackupLink(Document document,
             AccountingXmlDocumentBackupLink accountingXmlDocumentBackupLink) throws IOException {
+        if (StringUtils.isBlank(accountingXmlDocumentBackupLink.getCredentialGroupCode())) {
+            LOG.error("createAttachmentFromBackupLink, the Credential Group Code is blank");
+            throw new IOException("Unable to download attachment with blank Credential Group Code: " + accountingXmlDocumentBackupLink.getLinkUrl());
+        }
+        
```

Comment:
I think we should also update the isLinkUrlValidForGroupCode.  If there are webservice credential group codes for for a group code, the function returns true.  I think iti should return false.  The LOG message in that code area should probably be an error now as well.

Reply:
I want to clarify what you meant by your requested isLinkUrlValidForGroupCode change. Do you want the method to return false when a group code does NOT have any web service credentials?

Reply:
Yes, I think that would be appropriate, and update the logging statement that already exists